src/test/ui/associated-type-bounds/bad-bounds-on-assoc-in-trait.stderr

```diff
@@ -47,17 +41,7 @@ LL |     type C: Clone + Iterator<Item: Send + Iterator<Item: for<'a> Lam<&'a u8
    |                                                                                             ^^^^ `<<Self as Case1>::C as Iterator>::Item` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `<<Self as Case1>::C as Iterator>::Item`
-note: required by a bound in `Sync`
-  --> $SRC_DIR/core/src/marker.rs:LL:COL
-   |
-LL | / pub unsafe auto trait Sync {
-LL | |     // FIXME(estebank): once support to add notes in `rustc_on_unimplemented`
-LL | |     // lands in beta, and it has been extended to check whether a closure is
-LL | |     // anywhere in the requirement chain, extend it as such (#48534):
-...  |
-LL | |     // Empty
-LL | | }
-   | |_^ required by this bound in `Sync`
+   = note: consider using `std::sync::Arc<<<Self as Case1>::C as Iterator>::Item>`; for more information visit <https://doc.rust-lang.org/book/ch16-03-shared-state.html>
```

Comment:
This now seems to suggest to use `Arc<..::Item>` instead of `..::Item`. It might be useful to say `consider using Arc instead of Rc: ...`.

Reply:
But That's exactly what this is suggesting: we _didn't_ have an `Rc` here. Ideally we should be able to tell what context this error is being emitted on, but sadly `rustc_on_unimplemented` doesn't have that filtering today.